### PR TITLE
Decompile field_effect.s

### PR DESCRIPTION
**Pull request by PikalaxALT**
Fully decompile field_effect.s, except three functions which currently cannot be made to match verbatim.

Note: Also includes commits from #314 and #316.

### CODE REVIEW

File: src/daycare.c [hunk 1]
```diff
@@ -34,67 +36,29 @@ u8 Daycare_CountPokemon(struct BoxPokemon *daycare_data)
 }
 
 #ifndef ASDF
```
Comment (yenatch) [inline src/daycare.c]: ????

File: src/field_effect.c [hunk 2]
```diff
@@ -207,3 +613,2790 @@ bool8 FieldEffectActiveListContains(u8 id)
...
+    sub_807DE68();
+    FreeSpritePaletteByTag(GetSpritePaletteTagByPaletteNum(sprite->oam.paletteNum));
+    if (sprite->oam.affineMode != 0)
+    {
+        FreeOamMatrix(sprite->oam.matrixNum);
+    }
+    DestroySprite(sprite);
+}
+
+#undef NONMATCHING
```
Comment (yenatch) [inline src/field_effect.c]: ????

File: src/palette.c [hunk 3]
```diff
@@ -615,7 +615,7 @@ static u8 UpdateFastPaletteFade(void)
             if (b < b0)
                 b = b0;
 
-            gPlttBufferFaded[i] = r | (g << 5) | (b << 10);
+            gPlttBufferFaded[i] = RGB(r, g, b);
```
Comment (yenatch) [inline src/palette.c]: One day, when gPlttBufferFaded is `PlttData[] __attribute__((packed))`, this could be
```c
gPlttBufferFaded[i] = {r, g, b};
```

**Comment by yenatch** [comment]
`include/asm.inc.h 100644 → 100755`
`include/field_effect.h 100644 → 100755`
`include/gba/defines.h 100644 → 100755`
`include/global.h 100644 → 100755`
`include/menu.h 100644 → 100755`
`include/rom4.h 100644 → 100755`
`include/sprite.h 100644 → 100755`
`ld_script.txt 100644 → 100755`
`src/field_effect.c 100644 → 100755`
`src/menu.c 100644 → 100755`
`src/scrcmd.c 100644 → 100755`

File: src/daycare.c (hunk 1 above)
Comment (PikalaxALT) [inline src/daycare.c]: blame the original contributor for that

Comment (yenatch) [inline src/daycare.c]: what

File: src/field_effect.c (hunk 2 above)
Comment (PikalaxALT) [inline src/field_effect.c]: blame the previous contributor for that

File: src/daycare.c (hunk 1 above)
Comment (marijnvdwerf) [inline src/daycare.c]: Appears to be introduced in d0549706007b1a24ffd75ba40594e70dda3ee123

Comment (yenatch) [inline src/daycare.c]: thanks. can it be removed now this function is decompiled?

Comment (PikalaxALT) [inline src/daycare.c]: pushed, waiting for travis

File: src/field_effect.c (hunk 2 above)
Comment (yenatch) [inline src/field_effect.c]: if this line was old, wouldnt it have a `-` counterpart somewhere

Comment (PikalaxALT) [inline src/field_effect.c]: idk i thought i saw `#undef NONMATCHING` somewhere else in source. at any rate, this comment is obsoleted by d8f4d3da44d52ac56d93a41edb24cf0a803f5c67

File: include/gba/defines.h [hunk 4]
```diff
@@ -58,4 +58,7 @@
 
 #define WIN_RANGE(a, b) (((a) << 8) | (b))
 
+#define min(a, b) (a >= b ? a : b)
+#define max(a, b) (a <= b ? a : b)
```
Comment (marijnvdwerf) [inline include/gba/defines.h]: @yenatch I though the gba/* includes shouldn't be modified?

Comment (yenatch) [inline include/gba/defines.h]: i meant by clang-format

Comment (yenatch) [inline include/gba/defines.h]: that said youre right, this should not be in gba/defines. should be in global

File: src/record_mixing.c [hunk 5]
```diff
@@ -428,14 +429,20 @@ u8 sub_80B9BBC(u16 *a)
 #undef NONMATCHING
```
Comment (marijnvdwerf) [inline src/record_mixing.c]: Another undef

**Pull request merged**